nixos/modules/services/web-servers/authelia.nix

```diff
@@ -0,0 +1,181 @@
+        type = types.str;
+      };
+
+      jwtSecretFile = mkOption {
+        type = types.path;
+        default = null;
+        description = ''
+          Path to your JWT secret used during identity verificaiton.
+        '';
+      };
+
+      oidcIssuerPrivateKeyFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to your private key file used to encrypt OIDC JWTs.
+        '';
+      };
+
+      oidcHmacSecretFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to your HMAC secret used to sign OIDC JWTs.
+        '';
+      };
+
+      sessionSecretFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to your session secret. Only used when redis is used as session storage.
+        '';
+      };
+
+      storageEncryptionKeyFile = mkOption {
+        type = types.path;
+        default = null;
+        description = ''
+          Path to your storage encryption key.
+        '';
+      };
+
+      settings = mkOption {
+        description = ''
+          Your Authelia config.yml as a Nix attribute set.
+
+          https://github.com/authelia/authelia/blob/master/config.template.yml
+        '';
+        default = { };
+        type = types.submodule {
+          freeformType = format.type;
+          options = {
+            default_2fa_method = mkOption {
+              type = types.enum [ "" "totp" "webauthn" "mobile_push" ];
+              default = "";
+              example = "webauthn";
+              description = ''
+                Default 2FA method for new users and fallback for preferred but disabled methods.
+              '';
+            };
+
+            server = {
+              host = mkOption {
+                type = types.str;
+                default = "localhost";
+                example = "0.0.0.0";
+                description = ''
+              '';
+              };
+
+              port = mkOption {
+                type = types.port;
+                default = 9091;
+                description = ''
+              '';
+              };
+            };
+
+            log = {
+              level = mkOption {
+                type = types.enum [ "info" "debug" "trace" ];
+                default = "info";
+                example = "debug";
+                description = ''
+              '';
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+in
+{
+  options.services.authelia.instances = with lib; mkOption {
+    default = { };
+    type = types.attrsOf (types.submodule autheliaOpts);
+  };
+
+  config =
+    let
+      mkInstanceServiceConfig = instance: {
+        description = "Authelia authentication and authorization server";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
```

Comment:
Just as a request from an outsider... 

Since this is an authentication platform, can we add some hardening to the systemd unit?

Reply:
Of course.
The upstream doesn't have any but I don't see why we can't add some

https://github.com/authelia/authelia/blob/master/authelia.service

I'll grab some items from other nixos modules with hardened services